sw/airborne/modules/digital_cam/dc_ctrl_parrot_mykonos.c

```diff
@@ -0,0 +1,311 @@
+  dc_ctrl_parrot_mykonos.timer = 0;
+  dc_ctrl_parrot_mykonos.photo_nr = 0;
+  dc_ctrl_parrot_mykonos.autoshoot = 0;
+  dc_ctrl_parrot_mykonos.log_delay = 0;
+
+#ifndef SITL
+  int ret __attribute__((unused));
+  //ret = system("kk"); //No need kill original AP proceess since new  AP is already running and that killed original process
+
+  //TIP: With media-ctl much more can be done for the parameters of the camera only basics are set.
+  ret = system("media-ctl -l \'\"mt9f002 0-0010\":0->\"avicam.0\":0[1]\'");
+  ret = system("media-ctl -l \'\"avicam_dummy_dev.0\":0->\"avicam.0\":0[0]\'");//No bottomcam, used internaly in AP
+
+  ret = system("prestart dxowrapperd");
+  ret = system("prestart pimp"); // pimp = Parrot IMaging Process
+  //ret = system("pimpctl list-cameras"); //TODO look for 1 or more then define a variable to use or not
+#if DC_CTRL_PARROT_MYKONOS_STREAM_AT_STARTUP
+  dc_ctrl_parrot_mykonos_command(DC_CTRL_PARROT_MYKONOS_STREAM_START);
+#endif
+
+#else
+  //Start your local processes so simulated flight with sim shooting can be perfromed
+  //E.g. using a tile map interface of a pre-recorded stream or a 3D generated image
+  //FIXME: make a complete example, he, it's opensouce and involves your work indeed...
+  //ret = system("whateveryouwantotstartlocallyaddithere");
+#endif
+}
+
+void dc_ctrl_parrot_mykonos_periodic(void)
+{
+  //Nice 'n ugly use of True/False
+  if (dc_ctrl_parrot_mykonos.timer) {
+    dc_ctrl_parrot_mykonos.timer--;
+  } else {
+	  dc_ctrl_parrot_mykonos_command(DC_CTRL_PARROT_MYKONOS_SHOOT);
+  }
+  // test log delay if set
+  if (dc_ctrl_parrot_mykonos.log_delay) {
+#ifndef SITL
+    if (get_sys_time_msec() > dc_ctrl_parrot_mykonos.log_delay) { //FIXME: Could also happen in SITL...
+#endif
+#if DC_CTRL_PARROT_MYKONOS_LOG
+      dc_ctrl_parrot_mykonos_log_shot_position();
+#endif
+
+#if DC_CTRL_PARROT_MYKONOS_SYNC_SEND
+      dc_ctrl_parrot_mykonos_send_shot_position();
+#endif
+      // increment photo number
+      dc_ctrl_parrot_mykonos.photo_nr++;
+      // unset log delay
+      dc_ctrl_parrot_mykonos.log_delay = 0;
+#ifndef SITL
+    }
+#endif
+  }
+
+  // Common DC Periodic task
+  dc_periodic();
+}
+
+//FIXME muiltiple cams and same DC cam API do no go together,
+// so for the time being not used here, also one needs to add SITL options
+
+/* Command the Camera
+ * Intermidate function so the universal PPRZ Camra API can be used
+ * No need to change flightplan, script or otherwhise
+ * should work on both Fixedwing and Rotorcraft
+*/
+
+#ifdef SITL
+void dc_send_command(uint8_t cmd)
+{
+	 //Nothing yet, empty framework here so sim compiles
+	switch (cmd) {
+    default:
+      break;
+	}
+}
+#endif
+
+
+/* Execute the Shoot, Record and Stream commands */
+void dc_ctrl_parrot_mykonos_command(enum dc_ctrl_parrot_mykonos_status cmd)
+{
+  int ret __attribute__((unused));
+
+  dc_ctrl_parrot_mykonos.status = cmd;
+  switch (cmd) {
+    case DC_CTRL_PARROT_MYKONOS_RECORD_START:
+#ifndef SITL
```

Comment:
you should make a function or a macro that does the system calls or nothing when doing sim

Reply:
Yes, already stage 3 and 4 in the making with more changes.  A separate pull  was the original plan but already so much other branches changes etc. "rotting" in the repo since it kind of busy. And then another week goes by, and another. Still not even came around a PR for all the Parrot minidrones and that was last year October.. Well will find a way to have more frequent to the point PR's somehow

Reply:
Basically was rewriting the whole DC.h interface and have universal API and support for multiple cam with same DC routines for e.g. Distance and feedback this to the GCS GUI. Halfway the change realized the other changes will not be in master anytime soon if this would be added. A better flying ETECS Disco is needed for tests for the OBC2018 dynamic avoidance. So ripped out the new work on DC so Disco fixes would land up in master soon.